### DISCUSSION TITLE
can not get working SATA M.2

### DISCUSSION THREAD

**Discussion by dmproj** [CloverHackyColor/CloverBootloader#630] [Q&A]
system dell t3610
PCIe adapter with 1 slot for NVME
PCIe adapter with 2 slots for NVME and SATA

either adapter when is used with NVME works ok, 
but the adapter with 2 slots does not recognize SATA M.2 SSD's
so the question is if I need to preload a driver for SATA when PC starts, was looking for such a driver but couldn't find it
appreciate your help!
--------------------------
echo step1
connect -r
set -v efishellmode 1.1.2
map -u
echo step 2
load fs0:\EFI\CLOVER\drivers\UEFI\NvmExpressDxe.efi
echo step2
map -u
echo step3
fs1:\EFI\boot\bootx64.efi

**Comment by ThomasFreedman** [comment]
I have the same problem on a Lenovo M93 system. Booting clover on usb can see the SATA drives on the J Micron dual SATA PCIe card, so it has the drivers, but I'm having difficulty configuring clover to cascade boot the EFI on the SATA. The op sys is within a zfs filesystem but the EFI is on a standard FAT32 partition. 

I suspect the BOOTX64.EFI on the SATA is capable of accessing the operating system within the zfs pool to boot it. The zfs boot-pool on the M.2 SATA PCIe card I'm trying to boot was created by the TrueNAS installer.
